service/src/main/kotlin/fi/espoo/evaka/emailclient/IEmailMessageProvider.kt

```diff
@@ -9,12 +9,27 @@ import fi.espoo.evaka.invoicing.service.IncomeNotificationType
 import fi.espoo.evaka.messaging.MessageThreadStub
 import fi.espoo.evaka.shared.ChildId
 import fi.espoo.evaka.shared.domain.FiniteDateRange
+import org.jsoup.Jsoup
 
 data class EmailContent(
     val subject: String,
     val text: String,
     @org.intellij.lang.annotations.Language("html") val html: String
-)
+) {
+    companion object {
+        private val TOO_MANY_NEWLINES = Regex("\n{3,}")
+        fun fromHtml(subject: String, html: String) =
+            Jsoup.parseBodyFragment(html).let { doc ->
+                doc.select("hr").forEach { it.replaceWith(doc.createElement("p").text("-----")) }
+
+                EmailContent(
+                    subject,
+                    text = doc.wholeText().replace(TOO_MANY_NEWLINES, "\n\n").trim(),
+                    html = doc.html(),
```

Comment:
Because the raw string might not be exactly the same (e.g. if there are errors). I'd much rather use the HTML *as interpreted* by Jsoup so that the text and html are as close to each other as possible.